awswrangler/s3/_read_parquet.py

```diff
@@ -246,14 +249,25 @@ def _read_parquet_file(
         s3_additional_kwargs=s3_additional_kwargs,
         s3_client=s3_client,
     ) as f:
-        pq_file: Optional[pyarrow.parquet.ParquetFile] = _pyarrow_parquet_file_wrapper(
-            source=f,
-            coerce_int96_timestamp_unit=coerce_int96_timestamp_unit,
-        )
-        if pq_file is None:
-            raise exceptions.InvalidFile(f"Invalid Parquet file: {path}")
+        if schema and version.parse(pa.__version__) >= version.parse("8.0.0"):
+            table = pyarrow.parquet.read_table(
+                f,
+                columns=columns,
+                schema=schema,
+                use_threads=False,
+                use_pandas_metadata=False,
+                coerce_int96_timestamp_unit=coerce_int96_timestamp_unit,
+            )
+        else:
+            pq_file: Optional[pyarrow.parquet.ParquetFile] = _pyarrow_parquet_file_wrapper(
+                source=f,
+                coerce_int96_timestamp_unit=coerce_int96_timestamp_unit,
+            )
+            if pq_file is None:
+                raise exceptions.InvalidFile(f"Invalid Parquet file: {path}")
```

Comment:
In this scenario, we're throwing `InvalidFile` if `pq_file` is None. The `pq_file` variable is `None` when the size of the Parquet file is 0 bytes, as mentioned in `_pyarrow_parquet_file_wrapper`.

We should make sure that we are implementing the same behavior in the other branch of this `if` statement, where we're using `pyarrow.parquet_read_table`.

Reply:
Added, thanks